ecs/pod_creator.go

```diff
@@ -27,11 +31,235 @@ func NewBasicECSPodCreator(c cocoa.ECSClient, v cocoa.Vault) (*BasicECSPodCreato
 
 // CreatePod creates a new pod backed by AWS ECS.
 func (m *BasicECSPodCreator) CreatePod(ctx context.Context, opts ...*cocoa.ECSPodCreationOptions) (cocoa.ECSPod, error) {
-	return nil, errors.New("TODO: implement")
+
+	mergedPodCreationOpts := cocoa.MergeECSPodCreationOptions(opts...)
+	mergedPodExecutionOpts := cocoa.MergeECSPodExecutionOptions(mergedPodCreationOpts.ExecutionOpts)
+
+	if err := mergedPodCreationOpts.Validate(); err != nil {
+		return nil, errors.Wrap(err, "invalid pod creation options")
+	}
+
+	if err := mergedPodExecutionOpts.Validate(); err != nil {
+		return nil, errors.Wrap(err, "invalid pod execution options")
+	}
+
+	secrets := m.getSecrets(mergedPodCreationOpts)
+
+	err := m.createSecrets(ctx, secrets)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating secrets")
+	}
+
+	taskDefinition, err := m.exportPodCreationOptions(ctx, mergedPodCreationOpts, secrets)
+	if err != nil {
+		return nil, errors.Wrap(err, "translating the register task definition input to the correct type")
+	}
+
+	registerOut, err := m.client.RegisterTaskDefinition(ctx, taskDefinition)
+	if err != nil {
+		return nil, errors.Wrap(err, "registering task definition")
+	}
+
+	if registerOut.TaskDefinition == nil || registerOut.TaskDefinition.TaskDefinitionArn == nil {
+		return nil, errors.New("expected a task definition from ECS, but none was returned")
+	}
+
+	taskDef := cocoa.NewECSTaskDefinition().SetID(*registerOut.TaskDefinition.TaskDefinitionArn).SetOwned(true)
+
+	runTask := m.exportTaskExecution(mergedPodExecutionOpts, *taskDef)
+
+	runOut, err := m.client.RunTask(ctx, runTask)
+	if err != nil {
+		return nil, errors.Wrap(err, "running task")
+	}
+
+	if runOut.Failures != nil {
+		catcher := grip.NewBasicCatcher()
+		for _, failure := range runOut.Failures {
+			catcher.Add(errors.Errorf("task '%s': %s: %s\n", *failure.Arn, *failure.Detail, *failure.Reason))
```

Comment:
This can be simplified to `catcher.Errorf("task '%s': %s: %s\n", *failure.Arn, *failure.Detail, *failure.Reason)`